core/domain/feedback_jobs_one_off_test.py

```diff
@@ -0,0 +1,94 @@
+
+    def setUp(self):
+        super(PopulateMessageCountOneOffJobTest, self).setUp()
+        feedback_models.GeneralFeedbackThreadModel(
+            id='exp1.thread1', entity_id='exp1', entity_type='state1',
+            original_author_id='author', message_count=None,
+            status=feedback_models.STATUS_CHOICES_OPEN,
+            subject='subject', summary='summary', has_suggestion=False,
+            ).put()
+        feedback_models.GeneralFeedbackMessageModel(
+            id='exp1.thread1.1', thread_id='exp1.thread1', message_id=1,
+            author_id='author', text='message text').put()
+        feedback_models.GeneralFeedbackMessageModel(
+            id='exp1.thread1.2', thread_id='exp1.thread1', message_id=2,
+            author_id='author', text='message text').put()
+        feedback_models.GeneralFeedbackThreadModel(
+            id='exp2.thread2', entity_id='exp2', entity_type='state2',
+            original_author_id='author', message_count=None,
+            status=feedback_models.STATUS_CHOICES_OPEN,
+            subject='subject', summary='summary', has_suggestion=False,
+            ).put()
+        feedback_models.GeneralFeedbackMessageModel(
+            id='exp2.thread2.1', thread_id='exp2.thread2', message_id=1,
+            author_id='author', text='message text').put()
+        feedback_models.GeneralFeedbackThreadModel(
+            id='exp3.thread3', entity_id='exp3', entity_type='state3',
+            original_author_id='author', message_count=None,
+            status=feedback_models.STATUS_CHOICES_OPEN,
+            subject='subject', summary='summary', has_suggestion=False,
+            ).put()
+
+    def _run_one_off_job(self):
+        """Runs the one-off MapReduce job."""
+        job_id = (
+            feedback_jobs_one_off
+            .PopulateMessageCountOneOffJob.create_new())
+        feedback_jobs_one_off.PopulateMessageCountOneOffJob.enqueue(
+            job_id)
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+        self.process_and_flush_pending_tasks()
+        return feedback_jobs_one_off.PopulateMessageCountOneOffJob.get_output(
+            job_id)
+
+    def test_message_count_job(self):
+        thread1 = feedback_models.GeneralFeedbackThreadModel.get('exp1.thread1')
+        thread2 = feedback_models.GeneralFeedbackThreadModel.get('exp2.thread2')
+        thread3 = feedback_models.GeneralFeedbackThreadModel.get('exp3.thread3')
+        thread1.message_count = None
+        thread1.put()
+        thread2.message_count = None
+        thread2.put()
+        thread3.message_count = None
+        thread3.put()
+        self.assertEqual(thread1.message_count, None)
+        self.assertEqual(thread2.message_count, None)
+        self.assertEqual(thread3.message_count, None)
+        job_output = self._run_one_off_job()
+        self.assertEqual(len(job_output[0]), 17)
```

Comment:
Can you assert that job_output == expected value? Just comparing the len is too indirect.

Reply:
DONE